core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -165,21 +169,62 @@ contract PricelessPositionManager is FeePayer, AdministrateeInterface {
      ****************************************/
 
     /**
-     * @notice Transfers ownership of the caller's current position to `newSponsorAddress`.
+     * @notice Requests to transfer ownership of the caller's current position to `newSponsorAddress`. Once the request liveness is passed,
+     * the sponsor can execute the transfer.
+     * @dev The liveness length is the same as the withdrawal liveness.
+     */
+    function requestTransfer() public onlyPreExpiration() {
+        PositionData storage positionData = _getPositionData(msg.sender);
+        require(positionData.transferRequestPassTimestamp == 0);
+
+        // Make sure the proposed expiration of this request is not post-expiry.
+        uint256 requestPassTime = getCurrentTime().add(withdrawalLiveness);
+        require(requestPassTime <= expirationTimestamp);
+
+        // Update the position object for the user.
+        positionData.transferRequestPassTimestamp = requestPassTime;
+
+        emit RequestTransfer(msg.sender);
+    }
+
+    /**
+     * @notice After a passed transfer request (i.e., by a call to `requestTransfer` and waiting
+     * `withdrawalLiveness`), transfers ownership of the caller's current position to `newSponsorAddress`.
      * @dev Transferring positions can only occur if the recipient does not already have a position.
      * @param newSponsorAddress is the address to which the position will be transferred.
      */
-    function transfer(address newSponsorAddress) public onlyPreExpiration() {
+    function transferPassedRequest(address newSponsorAddress) public onlyPreExpiration() {
```

Comment:
Juts want to highlight again that I made a design decision to allow the sponsor to specify the `newSponsorAddress` on this call as opposed to when they initially make the `requestTransfer` call. I saw no downside to giving sponsor flexibility post-liveness and the upside is we don't have to store an `address` in `PositionData`.